netlogo-core/src/main/fileformat/PlotConverter.scala

```diff
@@ -5,36 +5,53 @@ package org.nlogo.fileformat
 import java.nio.file.Path
 
 import org.nlogo.core.{ CompilationEnvironment, Dialect, ExtensionManager,
-  LiteralParser, Model, Plot, Pen, SourceRewriter }
+  LiteralParser, Model, Plot, SourceRewriter }
 import org.nlogo.api.AutoConvertable
 
 object PlotConverter {
   def allPlotNames(model: Model): Seq[String] =
     model.plots.flatMap(_.display).filterNot(_.isEmpty)
 
-  def allPenNames(model: Model): Seq[String] =
-    model.plots.flatMap(_.pens).collect {
-      case pen: Pen => pen
-    }.map(_.display)
+  def allKeyedPenNames(model: Model): Seq[(String, Seq[String])] = {
+    model.plots
+      .map((plot: Plot) => (plot.display.getOrElse(""), plot.pens.toSeq.map(_.display)))
+      .filterNot(_._1.isEmpty)
+  }
 
-  private def determineRenames(names: Seq[String]): Seq[(String, String)] = {
-    names.groupBy(_.toUpperCase)
-      .toSeq
+  private[fileformat] def mapRenameDecision(groupedNames: Seq[(String, Seq[String])]):  Seq[(String, String)] = {
```

Comment:
The name here is ambiguous to me, since `map` might be a verb, but it might be a noun.  Either way, I think we can clarify to something more along the [naming guidelines](
https://github.com/NetLogo/NetLogo/wiki/Code-formatting#naming), like `generateRenames`?